### PR TITLE
Added completer for directories

### PR DESCRIPTION
**Pull request by drmalex07**
Hello, and congrats for the nice work. Lately, I have been using `argcomplete` more and more for my administrative Python scripts.

I sometimes came across the situation where i needed directory-only completion (e.g. like `complete -d`). I have created a completer class to address this, and i thought that it could be usefull to someone.

I did'nt use the subprocess approch that you have followed in `FilesCompleter`, because the task is quite simple and can be easily performed with plain `os.path` tools.

### CODE REVIEW

**Comment by kislyuk** [comment]
Looks good overall, please remove trailing semicolons from your lines.

**Comment by kislyuk** [comment]
Also, please add a test for your completer. You should be able to just copy/paste one of the existing tests and build a custom parser that uses this completer. To make it real, you might want to use https://docs.python.org/3.4/library/tempfile.html functions.

**Comment by drmalex07** [comment]
Hello @kislyuk. I 've removed the unneeded semicolons.

Regarding to the tests, i suppose we need a test similar to the one at
https://github.com/kislyuk/argcomplete/blob/master/test/test.py#L174 ??

**Comment by kislyuk** [comment]
Correct. Thanks.

**Comment by drmalex07** [comment]
Hello @kislyuk. I've added a relevant test. Please, check if this is adequate.

And one small question: Before adding this test, Travis failed at the phase of PyPy3 (while checking coverage). Has this failure anything to do with this, i mean are these tests strictly required for the coverage checks to pass ?

**Comment by kislyuk** [comment]
Thanks. I'll check this later today. Don't worry about the failure during
coverage checking - that is a bug in the test suite that got recently
introduced, and is not your fault. Sorry about that.

-ak

**Comment by drmalex07** [comment]
Hello @kislyuk , is there anything (from my side) pending on this issue ?

File: test/test.py [hunk 1]
```diff
@@ -183,6 +183,30 @@ def test_file_completion(self):
...
+            # Test completions
+            self.assertEqual(c('a'), set(['abb/', 'abc/']))
+            self.assertEqual(c('ab'), set(['abc/', 'abb/']))
+            self.assertEqual(c('abc'), set(['abc/']))
+            self.assertEqual(c('abc/'), set(['abc/baz/', 'abc/faz/']))
+            self.assertEqual(c('d'), set(['def/']))
+            self.assertEqual(c('def/'), set(['def/baz/']))
+            self.assertEqual(c('e'), set([]))
+            self.assertEqual(c('def/k'), set([]))
+        return
```
Comment (kislyuk) [inline test/test.py]: The `return` is unnecessary, please remove it.

**Comment by kislyuk** [comment]
Sorry about the delay, thanks for your patience @drmalex07. I went ahead and merged this. I'll try to roll a release soon.

**Comment by drmalex07** [comment]
No problem at all, i only asked if anything was missing. 

I am pretty sure we all have pressing deadlines on our heads!

**Pull request merged**